### DISCUSSION TITLE
removing the .git folder causes build to fail due to lack of git tags

### DISCUSSION THREAD

**Discussion by nikolarobottesla** [thliebig/openEMS-Project#28] [Q&A Usage] [answered]
@thliebig Is this the intended behavior? 

This also means forks need to have the tags to build. If they have some tags but don't have the latest tags (github doesn't sync them automatically after the initial fork. Also if you select 'master only' when you create the fork you don't get any tags) thus the build will have the wrong version.

**Comment by thliebig** [comment, marked as answer]
Yes that is intended behavior as I want to always see and know with which git revision a version was build.
And I really don't know why you would want to delete the history or not use git to continue to track your changes.

best regards
Thorsten

**Reply by thliebig** [reply]
I guess one could argue that the build should not fail if git is not available. But this was never something I thought about...

**Reply by nikolarobottesla** [reply]
Also when the build fails due to a lack of tags or .git folder, the error in the main log is not helpful. I haven't figured out how to get the /tmp/tmp.* log yet. It seems to not be available at the specified path immediately after the build fails.

By default github has the 'master only' box checked when you fork so anyone who forks with default settings won't get the tags, will have trouble building and because there is no error about missing tags, they won't know why the build failed.

**Reply by thliebig** [reply]
Feel free to supply a pull request with a patch that if this doesn't work still continues to build? ;)

**Comment by biergaizi** [comment]
Here's my recipe for working on a out-of-tree fork of openEMS, if I recall correctly, using this method, no tags are needed.

```
# clean upstream source
cd ~/
git clone --recurse-submodule https://github.com/thliebig/openEMS-Project.git

# change openEMS submodule URL
cd ~/openEMS-Project
git submodule set-url -- openEMS https://github.com/your/repo.git
git submodule sync

# update openEMS submodule
cd openEMS
git pull
git checkout your-branch
```

**Reply by thliebig** [reply]
well you have git in working order this way. Thus the build will get a valid tag and sha1. This is how it is supposed to work. @nikolarobottesla tried to remove everything git and that did not work, but maybe it should.
